packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -154,7 +146,7 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
   const isApiRoute = createApiRoutes(apiRoutes);
   const defaultAfterAuth = createDefaultAfterAuth(isPublicRoute, isApiRoute, params);
 
-  telemetry.record('METHOD_CALLED', {
+  clerkClient.telemetry.record('METHOD_CALLED', {
```

Comment:
☁️ I think we should not depend on the `clerkClient` as customers may have initialized and used a different clerk instance to authenticate their requests. We should record those telemetry metrics in that clerk instance and not on the default clerk instance.

Reply:
The dependency is fairly loose here, it was a convenient spot to piggyback off of the PK and SK they are providing to the backend client. Alternatively we could have each SDK instantiate its own telemetry instance.

Ultimately the SK and PK that we are using here isn't necessarily critical...it's more important that the SDK attribution is correct I think.

Reply:
I thought the attribution was based on the PK / SK to determine the instance_id. If it's not, then ignore my previous comment. 
It's about correct attribution and not about making each SDK instantiate its own telemetry. In the current line there may not be a valid clerk client instantiate since we may instantiate the clerk client during the `authenticateRequest` step.

Reply:
> I thought the attribution was based on the PK / SK to determine the instance_id.

@dimkl This is true, but what I'm saying is it's not absolutely critical. Telemetry is an internal concern, so my current thinking is that the most reliable approach we can take is to use the PK/SK that are provided through the recommended environment variables. We want a best-effort attribution to the PK/SK the application is using.